packages/ebayui-core-react/src/ebay-dialog-base/components/dialogBase.tsx

```diff
@@ -98,7 +98,7 @@ export const DialogBase: FC<DialogBaseProps<HTMLElement>> = ({
     useEffect(() => {
         let timeout: number
         const handleBackgroundClick = (e: DocumentEventMap['click']) => {
-            if (drawerBaseEl.current && !drawerBaseEl.current.contains(e.target)) {
+            if (drawerBaseEl.current && !drawerBaseEl.current.contains(e.target) && document.contains(e.target as Node)) {
```

Comment:
Can we add a comment for why we are adding document.contains. Also, I think you can check by `e.target.parentNode`, if it is `null` or empty it means its not on the dom and might be faster then `contains`